base/leaky_bucket.go

```diff
@@ -43,10 +43,21 @@ type LeakyBucketConfig struct {
 
 func NewLeakyBucket(bucket Bucket, config LeakyBucketConfig) Bucket {
 
-
-	_, ok := bucket.(*LeakyBucket)
+	leakyBucket, ok := bucket.(*LeakyBucket)
 	if ok {
-		panic(fmt.Sprintf("Cannot wrap bucket in a LeakyBucket, would be double wrapping: %+v", bucket))
+		// This case happens on TestSkippedViewRetrieval() because the GetBucket() will use a LeakyBucket based on the
+		// config, but then later in the code path, it calls NewLeakyBucket() with that LeakyBucket.
+		// To avoid test failure, just set the LeakyBucketConfig to the param of this function, but discard the
+		// previous leaky bucket config.
+		//
+		// Better approaches:
+		// - Avoid double-wrapping (which would probably require all tests that depend on GetBucket() to return a leaky bucket, and be more explicit about it)
+		// - Merge the leaky bucket configs rather than discarding the original config.
+
+		// This could merge config if we had a way to do that
+		Warnf(KeyAll, "Avoid double-wrapping LeakyBucket.  Discard original config and use incoming config");
+		leakyBucket.config = config
+		return leakyBucket
```

Comment:
As discussed, would be nice to prevent this in the first place by changing how the tests are creating a leaky bucket (i.e. changing setupTestLeakyDBWithCacheOptions)